lib/moon/packages/data_bags/data_matrix.rb

```diff
@@ -152,7 +142,7 @@ def [](x, y, z)
       y = y.to_i
       z = z.to_i
       return @default unless contains?(x, y, z)
-      @data[calc_index(x, y, z)]
+      @data[calc_index(x, y, z)] || @default
```

Comment:
@archSeer `contains?` won't fail it tests the xsize, ysize, zsize, not the data.size (you can't tell what dimension the data has from just .size alone), even the total size of the data is pre-calculated as `@size`